src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -1012,6 +1023,35 @@ private void ShowUpdatePackages(IVsWindowFrame windowFrame, ShowUpdatePackageOpt
             packageManagerControl?.ShowUpdatePackages(updatePackageOptions);
         }
 
+        public async Task ShowReadMeMarkdownToolWindowAsync(string filePath)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync(DisposalToken);
+
+            var toolWindow = FindToolWindow(typeof(ReadMeMarkdownToolWindow), 0, create: true);
+
+            if (!(toolWindow is ReadMeMarkdownToolWindow markdownToolWindow))
+            {
+                return;
+            }
+
+            await markdownToolWindow.UpdateAsync(filePath);
+
+            Guid persistenceSlot = typeof(ReadMeMarkdownToolWindow).GUID;
+
+            IVsUIShell vsUIShell = (IVsUIShell)GetService(typeof(SVsUIShell));
+            if (vsUIShell != null)
```

Comment:
echoing zivkan's comment, I suggest removing this check.
I'm not sure if explicitly throwing makes sense.
OR, if you can assume it won't be null since we do that [in other parts of this file](https://github.com/NuGet/NuGet.Client/blob/3cbda9cf4012c156a2b03977f151814b131cb0c5/src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs#L415). @zivkan  thoughts?

Reply:
That'll throw an exception, rather than silently fail, so I approve.  Just make sure that the exception bubbled up to an exception handler that logs fault telemetry, which is what I was trying to get at with this comment: https://github.com/NuGet/NuGet.Client/pull/4156#discussion_r674916691.  Although if customers start complaining that their package installls keep failing because of a null reference exception in a `ShowReadme` method, then at least we'll know that something we consider impossible is happening.

Reply:
VS fails with HResult and other DTE issues fairly often. Not able to look right now, but almost certain those are logged in telemetry already.

I just think we shouldn't add explicit telemetry and code testing for something like a fatal VS service call.... that's not really our fault/problem and not actionable by our extension. These should be visible in VSFeedback activity logs or similar, I'd think. If not, we can file a request against the team responsible for that (VS Platform?). I agree, testing that would be a good idea to see the behavior and logs generated.

Reply:
I don't agree that COM HResult failures are probably logged already. I think the cost of trying to investigate silent failures is so high that we should err on the side of double report errors, and remove one of the error logging once we validate that the error is already logged elsewhere.